common/unsafe/src/main/java/org/apache/spark/unsafe/array/LongArray.java

```diff
@@ -69,8 +64,9 @@ public long size() {
    * Fill this all with 0L.
    */
   public void zeroOut() {
+    long baseOffset = memory.getBaseOffset();
     for (long off = baseOffset; off < baseOffset + length * WIDTH; off += WIDTH) {
```

Comment:
the `off` should starts with 0.

Reply:
Good catch. These three unchanges seems to lead to failures.